api/service/projects_service.go

```diff
@@ -116,35 +124,56 @@ func (service *projectsService) ListProjects(ctx context.Context, name string, u
 	return allProjects, nil
 }
 
-func (service *projectsService) UpdateProject(ctx context.Context, project *models.Project) (*models.Project, error) {
+func (service *projectsService) UpdateProject(ctx context.Context, project *models.Project) (*models.Project,
+	map[string]interface{}, error) {
 	if service.authEnabled {
 		err := service.updateAuthorizationPolicy(ctx, project)
 		if err != nil {
-			return nil, fmt.Errorf("error while updating authorization policy for project %s", project.Name)
+			return nil, nil, fmt.Errorf("error while updating authorization policy for project %s", project.Name)
 		}
 	}
-	if service.webhookManager == nil || !service.webhookManager.IsEventConfigured(ProjectUpdatedEvent) {
-		return service.save(project)
+
+	areBlacklistedLabelsChanged, err := service.areBlacklistedLabelsChanged(project.ID, project.Labels,
+		service.updateProjectConfig.LabelsBlacklist)
```

Comment:
In general, if your method's going to take in a parameter that's actually a field/attribute of the method's object itself, i.e. `self.do_something(self.something_about_myself)` in Python, you can actually refactor the method to not even take in that argument, because it's redundant:

```go
func (service *projectsService) areBlacklistedLabelsChanged(project *models.Project) (bool, error) {
	...
	if service.updateProjectConfig.LabelsBlacklist[existingLabel.Key] {
	...
```
And oh, you can probably refactor the signature of the method too to take in only `project` since both `project.ID` and `project.Labels` can be retrieved from `project`.